lib/config_module/config_option.rb

```diff
@@ -36,29 +49,47 @@ def method_missing name, *args, &block
           "Key not found: #{name}", caller(3)
         )
       end
-
     rescue NoMethodError => error
       raise(
         ConfigOption::NotFoundError.new(name, self, error),
         error.message, caller(3)
       )
     end
 
-    def respond_to_missing? name, include_all
-      @table.has_key?(name) || super
-    end
+    if private_instance_methods.include? :new_ostruct_member!
+      def new_ostruct_member! name
+        name = name.to_sym
+        unless singleton_class.method_defined? name
+          define_singleton_method(name) { self.class.wrap @table[name] }
+          define_singleton_method("#{name}=") { |val| modifiable[name] = val }
+        end
+        name
+      end
+      private :new_ostruct_member!
 
-    def new_ostruct_member name
-      name = name.to_sym
-      unless respond_to? name
-        define_singleton_method(name) { self.class.wrap @table[name] }
-        define_singleton_method("#{name}=") { |x| modifiable[name] = x }
+      alias new_ostruct_member new_ostruct_member! # :nodoc:
+      protected :new_ostruct_member
+    elsif instance_methods.include? :new_ostruct_member
+      # Override for Ruby
+      def new_ostruct_member name
+        name = name.to_sym
+        unless respond_to? name
+          define_singleton_method(name) { self.class.wrap @table[name] }
+          define_singleton_method("#{name}=") { |val| modifiable[name] = val }
+        end
+        name
       end
-      name
+      protected :new_ostruct_member
     end
 
     class NotFoundError < ::ConfigModule::ConfigError
       def identifier; :key; end
     end
+
```

Comment:
Feels weird to have the above error class definition in the middle of a bunch of method definitions. Maybe it should just go at the top?